src/main/java/com/commercetools/sync/customers/utils/CustomerUpdateActionUtils.java

```diff
@@ -0,0 +1,994 @@
+
+        if (newAddressKey != null) {
+            if (oldAddress == null || !Objects.equals(oldAddress.getKey(), newAddressKey)) {
+                return Optional.of(SetDefaultShippingAddressWitKey.of(newAddressKey));
+            }
+        } else if (oldAddress != null) { // unset
+            return Optional.of(SetDefaultShippingAddressWitKey.of(null));
+        }
+
+        return Optional.empty();
+    }
+
+    /**
+     * Compares the {@link Customer#getDefaultBillingAddress()} and {@link CustomerDraft#getDefaultBillingAddress()}.
+     * If they are different - return {@link SetDefaultShippingAddressWitKey} update action. If the old billing address
+     * id value is set, but the new one is empty - the command will unset the default billing address.
+     *
+     * @param oldCustomer the customer that should be updated.
+     * @param newCustomer the customer draft with new default billing address.
+     * @return An optional with {@link SetDefaultShippingAddressWitKey} update action.
+     */
+    @Nonnull
+    public static Optional<UpdateAction<Customer>> buildSetDefaultBillingAddressUpdateAction(
+        @Nonnull final Customer oldCustomer,
+        @Nonnull final CustomerDraft newCustomer) {
+
+        final Address oldAddress = oldCustomer.getDefaultBillingAddress();
+        final String newAddressKey =
+            getAddressKeyAt(newCustomer.getAddresses(), newCustomer.getDefaultBillingAddress());
+
+        if (newAddressKey != null) {
+            if (oldAddress == null || !Objects.equals(oldAddress.getKey(), newAddressKey)) {
+                return Optional.of(SetDefaultBillingAddressWitKey.of(newAddressKey));
+            }
+        } else if (oldAddress != null) { // unset
+            return Optional.of(SetDefaultBillingAddressWitKey.of(null));
+        }
+
+        return Optional.empty();
+    }
+
+    @Nullable
+    private static String getAddressKeyAt(
+        @Nullable final List<Address> addressList,
+        @Nullable final Integer index) {
+
+        if (index != null) {
+            if (addressList != null) {
+                if (index >= 0 && index < addressList.size()) {
+                    final Address address = addressList.get(index);
+                    if (address == null) {
+                        throw new IllegalArgumentException(
+                            format("Address is null at the index: %s of the addresses list.", index));
+                    } else if (isBlank(address.getKey())) {
+                        throw new IllegalArgumentException(
+                            format("Address does not have a key at the index: %s of the addresses list.", index));
+                    }
+
+                    return address.getKey();
+                }
+            }
+
+            throw new IllegalArgumentException(
+                format("Addresses list does not contain an address at the index: %s", index));
+        }
+
+        return null;
```

Comment:
Can be simplified like following
```suggestion
                if (addressList == null) {
            throw new IllegalArgumentException(
                    format("Addresses list does not contain an address at the index: %s", index));
        }

        if (index == null || (index < 0 && index >= addressList.size())) {
            return null;
        }

        final Address address = addressList.get(index);
        if (address == null) {
            throw new IllegalArgumentException(
                format("Address is null at the index: %s of the addresses list.", index));
        } else if (isBlank(address.getKey())) {
            throw new IllegalArgumentException(
                format("Address does not have a key at the index: %s of the addresses list.", index));
        }
        return address.getKey();
```

Reply:
What do you mean by that?

Reply:
It would be more readable if we implement fewer level of nested `if-else` checking. Therefore it is suggested to throw `IllegalArgumentException` at the beginning of the method, and so on.

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/591/commits/b9d0f43884dcd1b6329cece1a7e16975056a89a1